api/models/auxil/status_setter.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+
+from django.utils import timezone
+
+from api.models import Coordinator, Group
+from api.models.choices.status import CoordinatorStatus, Status
+from api.models.shared_abstract.person import Person
+
+
+class StatusSetter:
```

Comment:
hmm, this looks like it might suffer from the same issues as `GroupProcessor` originally did... i think an approach like that could work well with creating log events since there isnt much logic there, but with setting statuses, there are different things to take into account for each action AND for each entity type

Reply:
on the other hand, it would be nice to have a separate `StatusSetter` for each entity type to abstract away that logic and not bulk up the processor classes

Reply:
For now, we are pretty sure the coordinators' statuses will follow the same rules no matter what event is triggered. In the future, though, we will have to refine the filters so we don't update statuses of coordinators that no longer are with the project. Maybe I should create a TODO.

Reply:
ok, i think i see the thinking here 👍 my only concern is that when we get to things like putting coordinators on leave, banning them, etc, presumably the logic for that will be in a different place. i might be getting too far ahead though, it makes sense with the current state of the code and we can always refactor later